src/libmongoc/CMakeLists.txt

```
@@ -705,11 +714,11 @@ endif ()
 
 set (LIBRARIES
    ${SASL_LIBRARIES} ${SSL_LIBRARIES} ${SHM_LIBRARIES} ${RESOLV_LIBRARIES}
-   ${SNAPPY_LIBRARIES} ${ZLIB_LIBRARIES} ${MONGOC_ZSTD_LIBRARIES} Threads::Threads ${ICU_LIBRARIES} ${LIBMONGOCRYPT_LIBRARY}
+   ${SNAPPY_LIBRARIES} ${ZLIB_LIBRARIES} ${UTF8PROC_LIBRARIES} ${MONGOC_ZSTD_LIBRARIES} Threads::Threads ${ICU_LIBRARIES} ${LIBMONGOCRYPT_LIBRARY}
```

```suggestion
   ${SNAPPY_LIBRARIES} ${ZLIB_LIBRARIES} ${MONGOC_ZSTD_LIBRARIES} Threads::Threads ${ICU_LIBRARIES} ${LIBMONGOCRYPT_LIBRARY}
```

Only bundling `utf8proc` is currently supported in this PR (not using a system install of `utf8proc`).
`${UTF8PROC_LIBRARIES} ` and `${UTF8PROC_INCLUDE_DIRS}` are always empty. Suggest removing all references to both `${UTF8PROC_LIBRARIES} ` and `${UTF8PROC_INCLUDE_DIRS}` in this file.